shell/platform/darwin/ios/ios_platform_surface_gl.mm

```diff
@@ -0,0 +1,67 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/darwin/ios/ios_platform_surface_gl.h"
+#include "flutter/common/threads.h"
+#include "flutter/lib/ui/painting/resource_context.h"
+#include "flutter/shell/platform/darwin/ios/framework/Source/vsync_waiter_ios.h"
+#include "third_party/skia/include/core/SkSurface.h"
+#include "third_party/skia/include/gpu/GrBackendSurface.h"
+#include "third_party/skia/include/gpu/GrTexture.h"
+#include "third_party/skia/include/gpu/GrTypes.h"
+
+namespace shell {
+
+IOSPlatformSurfaceGL::~IOSPlatformSurfaceGL() {
+  // TODO(sigurdm, mravn): dispose cache
+}
+
+IOSPlatformSurfaceGL::IOSPlatformSurfaceGL(NSObject<FlutterPlatformSurface>* surface)
+    : surface_(surface) {
+  FXL_DCHECK(surface_);
+}
+
+sk_sp<SkImage> IOSPlatformSurfaceGL::MakeSkImage(int width, int height, GrContext* grContext) {
+  ASSERT_IS_GPU_THREAD;
+  CVPixelBufferRef buffer;
+  fxl::AutoResetWaitableEvent latch;
+  blink::Threads::IO()->PostTask([this, &latch, &buffer]() {
```

Comment:
A few suggestions on getting rid of this switch:
- Will cause the GPU thread to block if there is an expensive decode job on the IO thread.
- The IO thread context is optional and may not be present at all.
- You are already on the GPU thread which contains an `EAGLContext`. This `EAGLContext` is different from the one on the IO thread but is in the same sharegroup. So use the `EAGLContext` on the GPU thread directly.
- It is actually better from a memory perspective if the texture cache is associated with the GPU thread `EAGLContext` because it will be purged during context loss on suspend. Not point in keeping cache items in the IO thread context which is not collected.